essentials-framework/Essentials Devices Common/Essentials Devices Common/VideoCodec/VideoCodecBase.cs

```diff
@@ -998,7 +1009,18 @@ private void LinkVideoCodecDirectoryToApi(IHasDirectory codec, BasicTriList tril
 
 			trilist.SetUShortSigAction(joinMap.DirectorySelectRow.JoinNumber, (i) => SelectDirectoryEntry(codec, i, trilist, joinMap));
 
-            // Report feedback for number of contact methods for selected contact
+            //Special Change for protected directory clear
+		    _directoryCodec = codec as IHasDirectoryClearSelection;
+
+		    if (_directoryCodec != null)
+		    {
+		        _directoryTrilist = trilist;
+		        _directoryJoinmap = joinMap;
+		        trilist.SetBoolSigAction(joinMap.DirectoryClearSelection.JoinNumber, (b) =>  DirectoryClearSelectionBase());

```

Comment:
Is there any reason this couldn't be a call do `SelectDirectoryEntry` directly? That would eliminate the need for storing the variables in private fields.

Reply:
Because SelectDirectoryEntry is a private method.  If I just directly called it, I wouldn't be able to clear the directory from within derived classes.  It was the whole reason I added this feature, otherwise you have to loop from a plugin out of a bridge to do anything to affect the directory entry selection.

I would **prefer** if SelectDirectoryEntry was protected - but changing method access levels is an larger API level change than just adding a new call that does a very specific thing.

Reply:
@andrew-welker - Give the changes we discussed a once-over.

I walked back most of the directory clear changes and moved most of the clear activity into the plugin.